File: padroesEstruturais/padrao-de-projeto-bridge/java/src/main/java/racas/Raca.java
package racas;

import classes.Classe;

public abstract class Raca {

    private Classe classe;
    public Raca(Classe classe) {
        this.classe = classe;
    }

    protected Classe getClasse() {
        return this.classe;
    }

    public abstract float calculaPoder();
}

File: padroesEstruturais/padrao-de-projeto-bridge/php/src/racas/Raca.php
<?php

namespace Bridge\Php\racas;

use Bridge\Php\classes\Classe;

abstract class Raca
{
    private Classe $classe;
    public function __construct(Classe $classe)
    {
        $this->classe = $classe;
    }

    protected function getClasse(): Classe
    {
        return $this->classe;
    }
    public abstract function calculaPoder(): float;
}